src/lib/core/rtl/directionality.ts

```diff
@@ -0,0 +1,18 @@
+import {EventEmitter, Injectable} from '@angular/core';
+import {LayoutDirection} from './dir';
+
+/**
+ * A provider to get the direction outside of angular on the `html` or `body` elements
+ */
+@Injectable()
+export class Directionality {
+  value: LayoutDirection;
+  change: EventEmitter<void> = new EventEmitter<void>();
+
+  constructor() {
+    this.value =
+      (document.body.getAttribute('dir') ||
```

Comment:
This should also check for the existence of `document` before dereferencing it (for platform-server)